flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterComponent.java

```diff
@@ -130,6 +131,13 @@ public T getDispatcher() {
 		}
 	}
 
+	@VisibleForTesting
+	public WebMonitorEndpoint<?> getWebMonitorEndpoint() {
```

Comment:
Coming back to my longer comment on #6749, I think this is only required because the collaborator is lazily created in a lifecycle method. I wonder if we could not have a building object the will create all collaborators at the place where currently `startComponent` is called because `clusterComponent = createClusterComponent(configuration)` and `clusterComponent.startComponent(...)` are even called in a straight sequence. Tests could then simply pass WebMonitorEndpoint into the component.

Reply:
Yes you're right. I've incorporated your feedback from #6743 and with that it should no longer be necessary to have this getter.